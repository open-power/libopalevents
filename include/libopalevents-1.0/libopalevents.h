/*
 *  libopalevents - OPAL Event parsing and printing library
 *  Copyright (C) 2015 IBM Corporation

 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.

 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.

 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

#ifndef _H_OPAL_EVENTS
#define _H_OPAL_EVENTS

#include <inttypes.h>

enum opal_event_log_flags {
	OPAL_EVENT_ALL = 0,
	OPAL_EVENT_ACTION = 1, /* Mutally exclusive */
	OPAL_EVENT_HEALTH = 2, /* Mutally exclusive */
	OPAL_EVENT_REPORT_ALL = 4,
	OPAL_EVENT_REPORT_HMC = 8,
	OPAL_EVENT_CALL_HOME = 16
};

typedef struct opal_event_log opal_event;

typedef enum opal_event_log_flags opal_event_flags;

opal_event *opal_event_create(void);

int opal_event_parse(opal_event *event, char *buf, int buflen);

int opal_event_print(opal_event *event);

int opal_event_summarise_hdr(void);

int opal_event_summarise(opal_event *event, opal_event_flags flags);

uint32_t opal_event_get_logid(opal_event *event);

int opal_event_destroy(opal_event *event);

#endif /* _H_OPAL_EVENTS */
