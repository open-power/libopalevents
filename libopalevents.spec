Name:           libopalevents
Version:        1
Release:        0%{?dist}
Summary:        PEL structure reader and parser

License:        LGPLv2
URL:            https://github.com/open-power/libopalevents
Source0:        https://github.com/open-power/libopalevents/archive/master.tar.gz
BuildRequires: autoconf automake libtool 
#Requires:       

%description 
PEL structure reader and parser
library to aid in reading, parsing and printing platform event log structures
 from files.



%package        devel
Summary:        Development files for %{name}
Requires:       %{name}%{?_isa} = %{version}-%{release}

%description    devel
The %{name}-devel package contains libraries and header files for
developing applications that use %{name}.


%prep

%setup -qn %{name}-master
autoreconf -if

%build
%configure --disable-static --prefix=/usr
make %{?_smp_mflags}


%install
rm -rf $RPM_BUILD_ROOT
%make_install
find $RPM_BUILD_ROOT -name '*.la' -exec rm -f {} ';'


%post -p /sbin/ldconfig

%postun -p /sbin/ldconfig


%files
%doc
%{_libdir}/*.so.*

%files devel
%doc
%{_includedir}/*
%{_libdir}/*.so
%{_libdir}/pkgconfig/libopalevents.pc


%changelog
* Wed Apr  8 2015 Cyril Bur <cyril.bur@au1.ibm.com> - 1
- Inital verion of the package
