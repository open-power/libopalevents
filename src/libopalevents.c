/*
 *  libopalevents - OPAL Event parsing and printing library
 *  Copyright (C) 2015 IBM Corporation

 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.

 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.

 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

/*
 *  libopalevents - OPAL Event parsing and printing library
 *  Copyright (C) 2015 IBM Corporation

 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.

 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.

 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */

#include <inttypes.h>

#include "libopalevents-1.0/libopalevents.h"
#include "opal-event-log.h"
#include "parse-opal-event.h"
#include "print-opal-event.h"

opal_event *opal_event_create(void)
{
	return create_opal_event_log();
}

int opal_event_parse(opal_event *event, char *buf, int buflen)
{
	return parse_opal_event_log(buf, buflen, event);
}

int opal_event_print(opal_event *event)
{
	return print_opal_event_log(event);
}

int opal_event_summarise_hdr(void)
{
	return header_opal_event_log();
}

int opal_event_summarise(opal_event *event, opal_event_flags flags)
{
	return summarise_opal_event_log(event, flags);
}

uint32_t opal_event_get_logid(opal_event *event)
{
	return opal_event_log_id(event);
}

int opal_event_destroy(opal_event *event)
{
	return free_opal_event_log(event);
}
