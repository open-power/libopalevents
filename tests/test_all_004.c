#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>
#include "libopalevents-1.0/libopalevents.h"
#include "test.h"
#include "test_fns.h"

int main(int argc, char *argv[])
{
	TEST the_t = {
		.name = "test_all_004 (5055ed2e)",
		.t = &test_all_normal,
		.cookie = (void *) 0,
		.r = { "test_all_004.out", "test_all_004.err" },
		.i = { "test_all_004.raw" }
	};

	return test_main(the_t);
}
