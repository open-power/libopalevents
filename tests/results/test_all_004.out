|------------------------------------------------------------------------------|
|                                Private Header                                |
|------------------------------------------------------------------------------|
| Section Version          : 1 (PH)                                            |
| Sub-section type         : 0x0                                               |
| Section Length           : 0x30                                              |
| Component ID             : 9500                                              |
| Created at               : 2014-02-18 | 06:43:54                             |
| Committed at             : 2014-02-18 | 06:43:54                             |
| Created by               : Service Processor                                 |
| Creator Sub Id           : 0x0 (0), 0x0 (0)                                  |
| Platform Log Id          : 0x5055ed2e                                        |
| Entry ID                 : 0x5055ed2e                                        |
| Section Count            : 12                                                |
|------------------------------------------------------------------------------|
|                                 User Header                                  |
|------------------------------------------------------------------------------|
| Section Version          : 1 (UH)                                            |
| Sub-section type         : 0x0                                               |
| Section Length           : 0x18                                              |
| Component ID             : 9500                                              |
| Subsystem                : Hypervisor firmware                               |
| Event Scope              : Single platform                                   |
| Event Severity           : Informational Event                               |
| Event Type               : Miscellaneous, informational only.                |
| Action Flags             : Healthy System Event                              |
|                          : No Service Action Required                        |
|------------------------------------------------------------------------------|
|                        Primary System Reference Code                         |
|------------------------------------------------------------------------------|
| Section Version          : 1 (PS)                                            |
| Sub-section type         : 0x1                                               |
| Section Length           : 0x50                                              |
| Component ID             : 9500                                              |
| SRC Format               : 0x0                                               |
| SRC Version              : 0x2                                               |
| Valid Word Count         : 0x9                                               |
| SRC Length               : 48                                                |
| Primary Reference Code   : B182950C                                          |
| Hex Words 2 - 5          : 020000F0 2B2C0E10 C103A401 000000FF               |
| Hex Words 6 - 9          : 00000042 E8CA2C00 00000000 00000000               |
|                                                                              |
|------------------------------------------------------------------------------|
|                             Extended User Header                             |
|------------------------------------------------------------------------------|
| Version                  : 1 (EH)                                            |
| Sub-section type         : 0                                                 |
| Component ID             : 3100                                              |
| Machine Type Model       : 8246-L2D                                          |
| Serial Number            : 060E8EA                                           |
| FW Released Ver          : ZL770_057                                         |
| FW SubSys Version        : b1126p_1320.770                                   |
| Common Ref Time (UTC)    :    0-00-00 | 00:00:00                             |
| Symptom Id Len           : 20                                                |
| Symptom Id               : B182950C_2B2C0E10                                 |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 2 (UD)                                            |
| Sub-section type         : 0x4                                               |
| Section Length           : 0x9c                                              |
| Component ID             : 3100                                              |
| User data hex            : length 148                                        |
|   00000000    00000b90  2f6f7074  2f666970  732f6269       ..../opt/fips/bi  |
|   00000010    6e2f6d62  6f786d61  696e7072  6f636573       n/mboxmainproces  |
|   00000020    73000000  00000000  00000000  00000000       s...............  |
|   00000030    00000000  00000000  00000000  00000000       ................  |
|   00000040    00000000  00000000  00000000  00000000       ................  |
|   00000050    00000000  00000000  66697073  3737302f       ........fips770/  |
|   00000060    62313132  36705f31  3332302e  37373000       b1126p_1320.770.  |
|   00000070    00000000  00000000  00000001  00000002       ................  |
|   00000080    00000804  00000004  00000006  00e51f64       ...............d  |
|   00000090    00000000                                     ....              |
|------------------------------------------------------------------------------|
|                      Machine Type/Model & Serial Number                      |
|------------------------------------------------------------------------------|
| Section Version          : 1 (MT)                                            |
| Sub-section type         : 0x0                                               |
| Section Length           : 0x1c                                              |
| Component ID             : 3100                                              |
| Machine Type Model       : 8246-L2D                                          |
| Serial Number            : 060E8EA                                           |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 1 (UD)                                            |
| Sub-section type         : 0xc                                               |
| Section Length           : 0x6c                                              |
| Component ID             : 3100                                              |
| User data hex            : length 100                                        |
|   00000000    01280342  4d424f58  4a000000  00000000       .(.BMBOXJ.......  |
|   00000010    00000000  00000064  00000000  00000064       .......d.......d  |
|   00000020    0000039c  00000018  00000026  973f3ab2       ...........&.?:.  |
|   00000030    00000bc7  0020434f  c3e6e01c  00000155       ..... CO.......U  |
|   00000040    0000000e  0000950c  00000042  e8ca2c00       ...........B..,.  |
|   00000050    00000000  00000000  00009510  5055ed2e       ............PU..  |
|   00000060    0000003c                                     ...<              |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 5 (UD)                                            |
| Sub-section type         : 0x1                                               |
| Section Length           : 0x24                                              |
| Component ID             : 9500                                              |
| User data hex            : length 28                                         |
|   00000000    00000017  0000004a  0000000c  00000042       .......J.......B  |
|   00000010    00000001  43413034  30303031                 ....CA040001      |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 5 (UD)                                            |
| Sub-section type         : 0x2                                               |
| Section Length           : 0x18                                              |
| Component ID             : 9500                                              |
| User data hex            : length 16                                         |
|   00000000    000000e8  000000ca  00000000  0000002c       ...............,  |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 12 (UD)                                           |
| Sub-section type         : 0x0                                               |
| Section Length           : 0xc                                               |
| Component ID             : 9500                                              |
| User data hex            : length 4                                          |
|   00000000    00000023                                     ...#              |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 1 (UD)                                            |
| Sub-section type         : 0xc                                               |
| Section Length           : 0x50                                              |
| Component ID             : 3100                                              |
| User data hex            : length 72                                         |
|   00000000    01280342  4d424f58  00000000  00000000       .(.BMBOX........  |
|   00000010    00000000  00000048  00000000  00000048       .......H.......H  |
|   00000020    00000640  0000002b  00000026  9730e378       ...@...+...&.0.x  |
|   00000030    00000bc8  0004434f  8dbf4c7f  000000ed       ......CO..L.....  |
|   00000040    5055ed2f  00000020                           PU./...           |
|------------------------------------------------------------------------------|
|                              User Defined Data                               |
|------------------------------------------------------------------------------|
| Section Version          : 1 (UD)                                            |
| Sub-section type         : 0xc                                               |
| Section Length           : 0x30                                              |
| Component ID             : 3100                                              |
| User data hex            : length 40                                         |
|   00000000    01280342  4d424f58  56000000  00000000       .(.BMBOXV.......  |
|   00000010    00000000  00000028  00000000  00000028       .......(.......(  |
|   00000020    000001d8  00000009                           ........          |
